README.md

```diff
@@ -114,20 +127,690 @@ validator-1  | [INFO  actix_web::middleware::logger] 127.0.0.1 "POST /finish HTT
 validator-1  | Hello world!
 ```
 
-Now you're ready to start building your Cartesi application with cartesi and deroll!
+## API and Usage Guide
+
+### 1. Create a new application
+
+#### `createApp(config: object): App`
+
+  Creates a new Cartesi application.
+
+  - **Parameters:**
+
+    - `config` (`object`): Configuration object.
+      - `url` (`string`): URL of the rollup HTTP server.
+
+  - **Returns:** `App` instance.
+
+  **Example:**
+
+  ```ts
+  import { createApp } from "@deroll/app";
+
+  const app = createApp({
+    url: process.env.ROLLUP_HTTP_SERVER || "http://127.0.0.1:5004",
+  });
+
+  app.start();
+  ```
+  ---
+
+### 2. Create notices, vouchers and reports
+
+#### `createNotice(request: { payload: 0x${string}; }): Promise<number>`
```

Comment:
This is actually a method of the `App` interface.
For a reference manual I'd rather have something generated.

Reply:
hi @tuler I pushed some changes

For the auto-generation, are we looking at something like TypeDoc? need some elaborate feedback there. thanks!